addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -709,6 +741,15 @@ registry.mediaVideo = publicWidget.Widget.extend(MobileYoutubeAutoplayMixin, {
             iframeEl = this._generateIframe();
         }
 
+        const sizeContainerEl = this.el.querySelector(":scope > .media_iframe_video_size");
+        if (iframeEl.dataset.needCookiesApproval) {
+            wUtils.setNoCookieSrcOrWarning(
+                iframeEl,
+                sizeContainerEl,
+                this._getPrivacyFriendlyVideoLink.bind(this)
+            );
+        }
```

Comment:
Adding the warning to all `needCookiesApproval` elements (except background videos) + the corresponding listener to remove it once the cookies have been approved, I guess that's feasible.
But each iframe's src should still be adapted and listen to the cookies approval. Some are created after the DOM is rendered, some (the videos) have a sizing container that should be shrunk down... In the end, we still need to call a function at some point, and that function needs to leave a listener to update the src, remove `.d-none`, etc.
So creating the warning at a generic place in the code wouldn't save lines, and it would actually split the logic / duplicate part of it in two separate places.

(I'm still thinking this one through, maybe I overlooked something. But that's my impression right now.)

Reply:
Nevermind my previous message, I got something to work. :)